src/Middleware/HttpLogging/src/W3CLoggerOptions.cs

```diff
@@ -0,0 +1,122 @@
+using Microsoft.Extensions.Hosting;
+
+namespace Microsoft.AspNetCore.HttpLogging
+{
+    /// <summary>
+    /// Options for the <see cref="W3CLogger"/>.
+    /// </summary>
+    public sealed class W3CLoggerOptions
+    {
+        private int? _fileSizeLimit = 10 * 1024 * 1024;
+        private int? _retainedFileCountLimit = 4;
+        private string _fileName = "w3clog-";
+        private string _logDirectory = "";
+        private TimeSpan _flushInterval = TimeSpan.FromSeconds(1);
+
+        /// <summary>
+        /// Gets or sets a strictly positive value representing the maximum log size in bytes or null for no limit.
+        /// Once the log is full, no more messages will be appended.
+        /// Defaults to <c>10MiB</c>.
+        /// </summary>
+        public int? FileSizeLimit
+        {
+            get { return _fileSizeLimit; }
+            set
+            {
+                if (value <= 0)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value), $"{nameof(FileSizeLimit)} must be positive.");
+                }
+                _fileSizeLimit = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a strictly positive value representing the maximum retained file count.
+        /// Defaults to <c>4</c>.
+        /// </summary>
+        public int? RetainedFileCountLimit
+        {
+            get { return _retainedFileCountLimit; }
+            set
+            {
+                if (value <= 0 || value > 10000)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value), $"{nameof(RetainedFileCountLimit)} must be between 1 and 10,000 (inclusive)");
+                }
+                _retainedFileCountLimit = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a string representing the prefix of the file name used to store the logging information.
+        /// The current date plus a file number (in the format {YYYYMMDD.X} will be appended to the given value.
+        /// Defaults to <c>w3clog-</c>.
+        /// </summary>
+        public string FileName
+        {
+            get { return _fileName; }
+            set
+            {
+                if (string.IsNullOrEmpty(value))
+                {
+                    throw new ArgumentNullException(nameof(value));
+                }
+                _fileName = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a string representing the directory where the log file will be written to
+        /// Defaults to <c>/logs/</c> relative to the app directory (ContentRoot).
+        /// </summary>
+        public string LogDirectory
+        {
+            get { return _logDirectory; }
+            set
+            {
+                if (string.IsNullOrEmpty(value))
+                {
+                    throw new ArgumentNullException(nameof(value));
+                }
+                if (!Path.IsPathRooted(value))
```

Comment:
Do we have any other APIs that accept a directory to do something? Do they expect a rooted path, or do they allow a relative path and prepend contentroot, but if you pass an absolute path they use it directly?

I'm worried that this isn't nice to use if you need to specify a full path because the full path often isn't known.

Found an example recently we did with the shadow copy stuff:
https://github.com/dotnet/aspnetcore/blob/2af293dc72625b865fa7fcad5aeb510df46386bd/src/Servers/IIS/AspNetCoreModuleV2/AspNetCore/applicationinfo.cpp#L265-L269

Reply:
Agreed. If this is a relative path, it should be relative to ContentRoot like the default `/logs/` is.